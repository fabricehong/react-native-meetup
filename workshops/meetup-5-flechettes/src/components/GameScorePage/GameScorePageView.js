import React, { PureComponent } from 'react';
import { View } from 'react-native';
import {
  Button,
  Container,
  Content,
  Footer,
  Form,
  Icon,
  Input,
  Item,
  Label,
  Left,
  List,
  ListItem,
  Right,
  Body,
  H1
} from 'native-base';

export class GameScorePageView extends PureComponent {
  renderResult() {
    playersList = [
      {
        name: 'Fabrice',
        score: 300
      },
      {
        name: 'Alpha',
        score: 299
      },
      {
        name: 'Arnaud',
        score: 12
      }
    ];

    const labels = playersList.map((player, index) => {
      const text = `${player.name}: ${player.score}`;
      return index === 0 ? <H1 key={index}>{text}</H1> : <Label key={index}>{text}</Label>
    });

    return (
      <View style={{ alignItems: 'flex-end' }}>
        {labels}
      </View>
    )
  }

  render() {
    return (
      <Container>
        <Content contentContainerStyle={{ flex: 1 }}>
          <Body style={{ justifyContent: 'center' }}>
            {this.renderResult()}
          </Body>
        </Content>
      </Container>
    );
  }
}

export default GameScorePageView;