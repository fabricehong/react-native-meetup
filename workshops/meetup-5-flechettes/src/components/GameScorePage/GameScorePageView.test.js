import React from 'react';
import renderer from 'react-test-renderer';
import GameScorePageView from './GameScorePageView';

describe('GameScorePageView', () => {
  test('should renders correctly', () => {
    const tree = renderer.create(<GameScorePageView/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});