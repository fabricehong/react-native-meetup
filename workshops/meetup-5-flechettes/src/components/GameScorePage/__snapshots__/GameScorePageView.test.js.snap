// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameScorePageView should renders correctly 1`] = `<View />`;
